src/EditorFeatures/Core/Implementation/GoToDefinition/INavigableDefinitionProvider.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Navigation;
+
+namespace Microsoft.CodeAnalysis.Editor
+{
+    internal interface INavigableDefinitionProvider
+    {
+        /// <summary>
+        /// Find definitions for the symbol referenced or defined at the given position in the given document.
+        /// </summary>
+        Task<IEnumerable<INavigableItem>> FindDefinitionsAsync(Document document, int position, CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Find definitions for the given symbol in the context of the given project.
+        /// </summary>
+        Task<IEnumerable<INavigableItem>> FindDefinitionsAsync(Project project, ISymbol symbol, CancellationToken cancellationToken);
```

Comment:
Agreed, avoiding this API would require some trivial refactoring of GoToDefinitionHelpers.TryGotoDefinition and hence I didn't do it now. We need to decide if the eventual design will support only document/position based API and/or symbol based API and make all the result providers have identical API shape.